nltk/internals.py

```diff
@@ -49,7 +49,10 @@ def convert_regexp_to_noncapturing_parsed(parsed_pattern):
                 raise ValueError('Regular expressions with back-references are not supported: {0}'.format(pattern))
             res_data.append((key, value))
         parsed_pattern.data = res_data
-        parsed_pattern.pattern.groups = 1
+        try:
+            parsed_pattern.pattern.groups = 1
+        except AttributeError:
+            pass
```

Comment:
By swallowing the error, the parsed_pattern could be left in an inconsistent state. If there are capturing groups, ``parsed_pattern.pattern.groups`` will be more than 1, yet ``parsed_pattern.pattern.groupdict`` will be empty.

Instead of trying to fix the symptom (pattern.groups cannot be assigned), why not try finding a better solution to compile_regexp_to_noncapturing? IMHO, the current implementation knows too much about the internals of sre_parse.py. Whenever the implementation of sre_parse changes, as with Python 3.5, it could break this function.

Another approach is to question the need for this function. This seems to be used only in ``nltk.tokenize.RegexpTokenizer``. Instead of removing capture groups silently, why not just throw an exception if there are capture groups? Note that this is an API change and will require discussion with the NLTK community.